xExtension-YouTube/extension.php

```diff
@@ -0,0 +1,260 @@
+
+        $iframe = '<iframe class="youtube-plugin-video"
+                style="height: ' . $this->height . 'px; width: ' . $this->width . 'px;"
+                width="' . $this->width . '"
+                height="' . $this->height . '"
+                src="' . $url . '"
+                frameborder="0"
+                allowFullScreen></iframe>';
+
+        if ($this->showContent) {
+            $doc = new DOMDocument();
+            $doc->encoding = 'UTF-8';
+            $doc->recover = true;
+            $doc->strictErrorChecking = false;
+
+            if ($doc->loadHTML('<?xml encoding="utf-8" ?>' . $entry->content()))
+            {
+                $xpath = new DOMXPath($doc);
+
+                $titles = $xpath->evaluate("//*[@class='enclosure-title']");
+                $thumbnails = $xpath->evaluate("//*[@class='enclosure-thumbnail']/@src");
+                $descriptions = $xpath->evaluate("//*[@class='enclosure-description']");
+
+                $content = '<div class="enclosure">';
+
+                // We hide the title so it doesn't appear in the final article, which would be redundant with the RSS article title,
+                // but we keep it in the content anyway, so RSS clients can extract it if needed.
+                if ($titles->length > 0) {
+                    $content .= '<p class="enclosure-title" hidden>' . $titles[0]->nodeValue . '</p>';
+                }
+
+                // We hide the thumbnail so it doesn't appear in the final article, which would be redundant with the YouTube player preview,
+                // but we keep it in the content anyway, so RSS clients can extract it to display a preview where it wants (in article listing,
+                // by example, like with Reeder).
+                if ($thumbnails->length > 0) {
+                    $content .= '<p hidden><img class="enclosure-thumbnail" src="' . $thumbnails[0]->nodeValue . '" alt=""/></p>';
+                }
+
+                $content .= $iframe;
+
+                if ($descriptions->length > 0) {
+                    $content .= '<p class="enclosure-description">' . nl2br(htmlentities($descriptions[0]->nodeValue)) . '</p>';
+                }
+
+                $content .= "</div>\n";
+            }
+            else {
+                $content = $iframe . $entry->content();
+            }
+        }
+        else {
+            $content = $iframe;
+        }
+
+        return $content;
+    }
+
+    /**
+     * This function is called by FreshRSS when the configuration page is loaded, and when configuration is saved.
+     *  - We save configuration in case of a post.
+     *  - We (re)load configuration in all case, so they are in-sync after a save and before a page load.
+     */
+    public function handleConfigureAction(): void
+    {
+        $this->registerTranslates();
+
+        if (Minz_Request::isPost()) {
+            FreshRSS_Context::userConf()->yt_player_height = Minz_Request::paramInt('yt_height');
```

Comment:
@Alkarex as I am upgrading code blindly here, is `_attribute()` there correct way to set configuration?

https://github.com/FreshRSS/FreshRSS/blob/354a5e83888cebe89843b754d0936f3633323cef/app/Models/UserConfiguration.php#L129-L131